libfreerdp/core/gateway/rdg.c

```diff
@@ -298,21 +305,113 @@ static BOOL rdg_write_packet(rdpRdg* rdg, wStream* sPacket)
 	return TRUE;
 }
 
-static BOOL rdg_read_all(rdpTls* tls, BYTE* buffer, int size)
+static int rdg_socket_read(BIO* bio, BYTE* pBuffer, size_t size, UINT16 transferEncoding,
+                           size_t* transferEncodingRemainingCount)
+{
+	if (transferEncoding == HTTP_TRANSFER_ENCODING_IDENTITY)
+	{
+		return BIO_read(bio, pBuffer, size);
+	}
+	else if (transferEncoding == HTTP_TRANSFER_ENCODING_CHUNKED)
+	{
+		int status;
+
+		if (transferEncodingRemainingCount == NULL)
```

Comment:
shouldn't it be an assert ? If this condition becomes true it would be because we would done something wrong in our code when calling `rdg_socket_read`.

Reply:
right. will change that